### PR TITLE
CORDA-2509 - Updating release notes

### PR DESCRIPTION
**Pull request by fenryka**
Adding @gendal 's note on 4.0 vs 4.1 inter-op with 3.x

### CODE REVIEW

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -24,7 +26,28 @@ apps that benefit from them, *all* nodes in a compatibility zone must be upgrade
...
 account for your own schedule planning.
 
-.. contents::
+
+Important note about Corda’s compatibility, upgrade and interoperability guarantees
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Corda platform places great emphasis on ensuring compatibility across versions and the ability to
+upgrade/migrate data from previous versions to new versions. As a CorDapp developer or operator of a node,
+it is important to understand these guarantees and how to take advantage of them. Please see xxx for
```
Comment (gendal) [inline docs/source/release-notes.rst]: any idea what the xxx should be?  I put it in as a placeholder but I'm not sure where we document it.

File: docs/source/release-notes.rst [hunk 2]
```diff
@@ -24,7 +26,28 @@ apps that benefit from them, *all* nodes in a compatibility zone must be upgrade
...
+full information on Corda’s capabilities and platform commitments.
+
+For Corda 4, these capabilities will be delivered progressively, across the 4.0 release, which is
+available now, and the 4.1 release, which will follow in the coming weeks. Here is how the capability is split:
+
+  *Corda 4.0* delivers the full new suite of Platform version 4 features: Reference states, signature constraints,
+  and more. See here for a full run-down. The API is stable and you can start developing new apps with it
```
Comment (gendal) [inline docs/source/release-notes.rst]: Change "See here for a full run-down" to "These features are summarised below."

File: docs/source/release-notes.rst [hunk 3]
```diff
@@ -24,7 +26,28 @@ apps that benefit from them, *all* nodes in a compatibility zone must be upgrade
...
+  today and upgrading existing apps to take advantage of the new features today. In addition, you can take these
+  apps live for new deployments.
+
+  *Corda 4.1* will deliver the ability to support mixed networks of Corda 3.3, Corda Enterprise 3.2 and
+  Corda 4.1 nodes and the ability to upgrade such nodes, even those with pre-existing data in their vaults to Corda 4.1.
```
Comment (gendal) [inline docs/source/release-notes.rst]: (Realised there were some missing commas in my version). How about this: 

*Corda 4.1* will deliver the ability to support mixed networks of Corda 3.3, Corda Enterprise 3.2 and Corda 4.1 nodes, and the ability to upgrade such nodes, even those with pre-existing data in their vaults, to Corda 4.1.

File: docs/source/release-notes.rst (hunk 1 above)
Comment (fenryka) [reply docs/source/release-notes.rst]: liknking to versioning, not sure it's the best page, we don't really spell out the explicit garuntees as far as I can tell

**Comment by gendal** [comment]
31 jan changes lgtm

**Pull request merged**